source/Engine.cpp

```diff
@@ -1598,20 +1598,41 @@ void Engine::HandleKeyboardInputs()
 	Command keyDown = keyHeld.AndNot(oldHeld);
 	
 	++landKeyInterval;
-	if(oldHeld.Has(Command::LAND))
-		landKeyInterval = 0;
 	
+	// These commands must always be sent:
+	activeCommands |= keyHeld.And(Command::PRIMARY | Command::SECONDARY | Command::SCAN |
+		Command::FORWARD | Command::LEFT | Command::RIGHT | Command::BACK | Command::AFTERBURNER);
+
+	// Are any movement keys held?
+	bool movementKeys=keyHeld.Has(Command::AFTERBURNER | Command::BACK | Command::FORWARD |
+			Command::LEFT | Command::RIGHT);
+
+	// Were any movement keys held in the previous step?
+	bool oldMovementKeys=oldHeld.Has(Command::AFTERBURNER | Command::BACK | Command::FORWARD |
+			Command::LEFT | Command::RIGHT);
+
+	if(!movementKeys && oldMovementKeys)
+	{
+		// Player has released all movement keys since the last step, so
+		// reinstate autopilot commands.
+		activeCommands |= keyHeld.And(Command::JUMP | Command::BOARD | Command::LAND);
+
+		// Make sure we do not switch landing targets:
+		landKeyInterval = 9999;
+	}
+	else if(oldHeld.Has(Command::LAND))
+		// We're not reinstating commands, and the LAND command was given in the prior
+		// step.  Reset our counter of the number of 1/60ths of a second since the land
+		// key was released.
+		landKeyInterval = 0;
+
 	// Wait with actual jumping until the jump command is released.
```

Comment:
Might as well update this whole comment to be:
```c++
// If holding JUMP or rapidly pressing LAND, also send WAIT, to prevent
// the autopilot from initiating the jump or settling on the landing target.
```